diff --git a/django/http/response.py b/django/http/response.py
index 6a84e193ba..cff154d008 100644
--- a/django/http/response.py
+++ b/django/http/response.py
@@ -231,6 +231,8 @@ class HttpResponseBase:
         # - when self._charset != 'utf-8' it re-encodes the content
         if isinstance(value, bytes):
             return bytes(value)
+        if isinstance(value, memoryview):
+            return value.tobytes()
         if isinstance(value, str):
             return bytes(value.encode(self.charset))
         # Handle non-string types.
diff --git a/tests/httpwrappers/tests.py b/tests/httpwrappers/tests.py
index e6c6297898..c478b3c92c 100644
--- a/tests/httpwrappers/tests.py
+++ b/tests/httpwrappers/tests.py
@@ -366,6 +366,10 @@ class HttpResponseTests(unittest.TestCase):
         r.content = 12345
         self.assertEqual(r.content, b'12345')
 
+    def test_memoryview_content(self):
+        r = HttpResponse(memoryview(b'My Content'))
+        self.assertEqual(r.content, b'My Content')
+
     def test_iter_content(self):
         r = HttpResponse(['abc', 'def', 'ghi'])
         self.assertEqual(r.content, b'abcdefghi')
